File: pedro-api/src/main/java/br/com/cpcx/web2/pedro/domain/entity/Pessoa.java
package br.com.cpcx.web2.pedro.domain.entity;

import br.com.cpcx.web2.pedro.domain.enumeration.ESituacaoPessoa;
import br.com.cpcx.web2.pedro.domain.enumeration.ETipoPessoa;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.DiscriminatorColumn;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.Table;
import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "TB_PESSOA")
@DiscriminatorColumn(name="PES_TIPO", length = 20)
@DiscriminatorValue("PESSOA")
@Inheritance(strategy = InheritanceType.JOINED)
public class Pessoa {

    @Id
    @Column(name = "PES_ID")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "PES_ID_RESPONSAVEL")
    private Long idResponsavel;

    @Enumerated(EnumType.STRING)
    @Column(name = "PES_TIPO", insertable = false, updatable = false)
    private ETipoPessoa tipo;

    @Enumerated(EnumType.STRING)
    @Column(name = "PES_SITUACAO")
    private ESituacaoPessoa situacao;

    @Column(name = "PES_NOME")
    private String nome;

    @Column(name = "PES_APELIDO")
    private String apelido;

    @Column(name = "PES_DATA_NASCIMENTO")
    private LocalDate dataNascimento;

}

File: pedro-app/src/models/pessoa/Pessoa.js
export default Pessoa;

class Pessoa{
    constructor(id, idResponsavel, tipoPessoa, situacaoPessoa, nome, apelido, cpf, cnpj, rg, dataNascimento) {
        this._id = id;
        this._idResponsavel = idResponsavel;
        this._tipoPessoa = tipoPessoa;
        this._situacaoPessoa = situacaoPessoa;
        this._nome = nome;
        this._apelido = apelido;
        this._cpf = cpf;
        this._cnpj = cnpj;
        this._rg = rg;
        this._dataNascimento = dataNascimento;

    }

    get id(){
        return this._id;
    }
    get idResponsavel(){
        return this._idResponsavel;
    }

    get tipoPessoa() {
        return this._tipoPessoa;
    }

    get situacaoPessoa() {
        return this._situacaoPessoa;
    }

    get nome() {
        return this._nome;
    }

    get apelido() {
        return this._apelido;
    }

    get cpf() {
        return this._cpf;
    }

    get cnpj() {
        return this._cnpj;
    }

    get rg() {
        return this._rg;
    }

    get dataNascimento() {
        return this._dataNascimento;
    }
}
